### DISCUSSION TITLE
Curious about performance for large number of pages

### DISCUSSION THREAD

**Discussion by dephiros** [ElMassimo/iles#75] [Q&A] [answered]
I am curious if anyone has tested the build time for a large number of pages(comparing to Astro, Hugo, or 11ty). I think the downside of a tool like VuePress(build with JS compiler) tends to be performance

**Comment by ElMassimo** [comment, marked as answer]
That's correct, there's a trade-off between expressiveness and performance.

For example, MDX allows you to use [shortcodes] to customize how the content is rendered, which means components can be provided dynamically. This provides maximum flexibility, at the expense of not being able to use a faster compiler with hard-coded output.

If you need to render a large amount (thousands) of pages, a tool like Hugo might be a better fit, as it's optimized for that use case.

For smaller sites, the difference in performance is not significant, and the ease of use and expressiveness might offset the difference.

---

If you have simple content (plain markdown or text) in the hundreds/thousands, a good way to obtain better build performance is to use a [dynamic route](https://iles-docs.netlify.app/guide/routing#dynamic-routes), which is a single file to compile, but can render many pages.

For example, you can create a "post" page, and then obtain content for hundreds of articles and render hundreds of pages without increasing the compilation time as you add more pages (only rendering time, which is orders of magnitude faster than compiling).

[shortcodes]: https://iles-docs.netlify.app/guide/markdown#advanced-override-elements-shortcodes
